packages/sdk-core/src/BacktraceCoreClient.ts

```diff
@@ -158,19 +172,38 @@ export abstract class BacktraceCoreClient {
         ).build();
 
         if (metrics) {
-            this._metrics = metrics;
+            this._modules.set(BacktraceMetrics, metrics);
         }
 
         if (this.options.breadcrumbs?.enable !== false) {
-            this.breadcrumbsManager = new BreadcrumbsManager(this.options?.breadcrumbs, this._setup.breadcrumbsSetup);
-            this._attachments.push(this.breadcrumbsManager.breadcrumbsStorage);
-            this.attributeManager.addProvider(this.breadcrumbsManager);
+            const breadcrumbsManager = new BreadcrumbsManager(this.options?.breadcrumbs, this._setup.breadcrumbsSetup);
+            this._attachments.push(breadcrumbsManager.breadcrumbsStorage);
+            this.attributeManager.addProvider(breadcrumbsManager);
+            this._modules.set(BreadcrumbsManager, breadcrumbsManager);
         }
 
-        this.initialize();
         this._enabled = true;
     }
 
+    public initialize() {
+        if (this.fileSystem && this.options.database?.createDatabaseDirectory) {
```

Comment:
I think this should be a part of the BacktraceDatabase implementation.

Reply:
This is really troublesome. If we didn't rely on a database directory for everything, but just use something like "storageOptions", this would make perfect sense.

Moving it to database will cause undeterministic behavior, when some module depending on DB's directory will be initialized before DB.